.github/workflows/ci.yml

```diff
@@ -712,3 +725,166 @@ jobs:
         with:
           name: ${{ env.artifacts_name }}
           path: artifacts
+
+  gaugerelease:
+    name: Check unreleased changes
+    runs-on: ubuntu-latest
+    needs: tests
+    # TODO: restricted to silverstripe/silverstripe-framework - remove once we've validated this whole thing works
+    if: ${{ github.repository == 'silverstripe/silverstripe-framework' && github.repository_owner == 'silverstripe' && (github.event_name == 'push' ||  github.event_name == 'schedule') }}
+    env:
+      GITHUB_REPOSITORY: ${{ github.repository }}
+      GITHUB_REF_NAME: ${{ github.ref_name }}
+      GITHUB_SHA: ${{ github.sha }}
+      LATEST_LOCAL_SHA: ${{ needs.tests.outputs.latest_local_sha }}
+    outputs:
+      do_release: ${{ steps.determine-if.outputs.do_release }}
+      release_notes: ${{ steps.determine-if.outputs.release_notes }}
+      latest_tag: ${{ steps.determine-if.outputs.latest_tag }}
+      next_tag: ${{ steps.determine-if.outputs.next_tag }}
+    steps:
+
+      - name: Determine if should patch release
+        id: determine-if
+        run: |
+          DO_RELEASE=1
+
+          # Must be on a minor branch to do a patch release
+          if ! [[ $GITHUB_REF_NAME =~ ^[0-9]+\.[0.9]+$ ]]; then
+            echo "Not patch releasing because not on a minor branch"
+            DO_RELEASE=0
+          fi
+
+          # Validate that this commit is that latest commit for the branch using GitHub API
+          # We need to check this in case re-rerunning an old job and there have been new commits since
+          if [[ $DO_RELEASE == "1" ]]; then
```

Comment:
Given the exit above, we don't need this condition anymore.

Reply:
I'd prefer to keep the variable rather than switch to a function